### PR TITLE
Added Storage Module

### PR DESCRIPTION
**Pull request by AliRKat**

### CODE REVIEW

**Comment by AliRKat** [comment]
[[NodeJS] Option to use memory only for storage](https://trello.com/c/3B9JMGQg/1943-nodejs-option-to-use-memory-only-for-storage)

File: lib/countly-storage.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
+const fs = require('fs');
+const path = require('path');
+var cc = require("./countly-common");
+var storagePath;
+var __data = {};
+
+var setStoragePath = function (path) {
+    defaultPath = "../data/"; // Default path
```
Comment (arifBurakDemiray) [inline lib/countly-storage.js:8]: what was the default path before?
was it ../bulk_data/

Comment (arifBurakDemiray) [reply lib/countly-storage.js:8]: well this is ok but countly and bulk-data defaults are different
If it is not a problem then all cool

**Comment by arifBurakDemiray** [comment]
And all things looks like failed, could you fix those

File: lib/countly-bulk.js [hunk 2]
```diff
@@ -105,7 +104,9 @@ function CountlyBulk(conf) {
     conf.maxStackTraceLinesPerThread = conf.max_stack_trace_lines_per_thread || maxStackTraceLinesPerThread;
     conf.maxStackTraceLineLength = conf.max_stack_trace_line_length || maxStackTraceLineLength;
 
-    var mainDir = path.resolve(__dirname, conf.storage_path);
+    CountlyStorage.setBulkDataPath(conf.storage_path);
+
+    var mainDir = path.resolve(__dirname, CountlyStorage.getStoragePath());
```
Comment (turtledreams) [inline lib/countly-bulk.js:109]: this and below should be carried over too

**Review by turtledreams — CHANGES_REQUESTED** [review]
Nice initial work
```
    var asyncWriteLock = false;
    var asyncWriteQueue = [];
```
should be resettable too. check Halt method. pls make sure linting is working. check vscode plugins are all fine. make sure tests work after fixing linting issues (use: npm run test , for tests)

**Review by arifBurakDemiray — APPROVED** [review]
adding tests would be great

**Pull request merged**